components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -504,7 +504,22 @@ private boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) throws I
             }
             return true;
         }
-
+        if (GrantType.REFRESH_TOKEN.toString().equals(grantType)) {
+            AuthorizationGrantHandler authzGrantHandler = authzGrantHandlers.get(grantType);
+            boolean isValidScope = authzGrantHandler.validateScope(tokReqMsgCtx);
```

Comment:
Why are we specifically calling the grant handler validate scope over here? isn't this anyway being called?

Reply:
No we are calling the grant handler specific validate scope methos and returning from here and omitting the extra validations. Because for refresh token we don't need to validate the scopes again with other scope validators since the scopes were already validated during the token issuing time.

Reply:
Let's add a debug log here to indicate this